be/src/exec/vectorized/partition/partition_hash_map.h

```diff
@@ -92,11 +106,20 @@ struct PartitionHashMapBase {
     template <typename HashMap, typename KeyLoader, typename KeyAllocator>
     void append_chunk_for_one_key(HashMap& hash_map, ChunkPtr chunk, KeyLoader&& key_loader,
                                   KeyAllocator&& key_allocator, ObjectPool* obj_pool) {
+        if (is_downgrade) {
+            return;
+        }
         phmap::flat_hash_set<typename HashMap::key_type, typename HashMap::hasher, typename HashMap::key_equal,
                              typename HashMap::allocator_type>
                 visited_keys(chunk->num_rows());
         const auto size = chunk->num_rows();
-        for (uint32_t i = 0; i < size; i++) {
+        uint32_t i = 0;
+        for (; i < size; i++) {
+            check_downgrade(hash_map);
```

Comment:
Would it hurt the performance? Check downgrade each row?

Reply:
Good point of view, I'll see if I can optimize it